lib/spack/spack/test/spack_yaml.py

```diff
@@ -88,7 +89,52 @@ def test_yaml_aliases():
         "e": aliased_list_2,
         "f": aliased_list_2,
     }
-    string = syaml.dump(dict_with_aliases)
+    stringio = io.StringIO()
+    syaml.dump(dict_with_aliases, stream=stringio)
 
     # ensure no YAML aliases appear in syaml dumps.
-    assert "*id" not in string
+    assert "*id" not in stringio.getvalue()
+
+
+@pytest.mark.parametrize(
+    "initial_content,expected_final_content",
+    [
+        # List are dumped indented as the outer attribute
+        (
+            """spack:
+  #foo
+  specs:
+  # bar
+  - zlib
+""",
+            None,
+        ),
+        (
+            """spack:
+  #foo
+  specs:
+    # bar
+    - zlib
+""",
+            """spack:
+  #foo
+  specs:
+    # bar
```

Comment:
Is this really how it dumps? with the comment indented but the elements dedented?

Reply:
Unfortunately yes. Comments are handled separately, and attached to yaml objects.